source/codegen/metadata/nidmm/functions.py

```diff
@@ -1624,7 +1624,7 @@
     },
     "InitExtCal": {
         "init_method" : True,
-        "codegen_method" : None,
+        "close_function" : "[&] (uint32_t id) { library_->CloseExtCal(id,NIDMM_EXTCAL_ACTION_ABORT); }",
```

Comment:
CloseExtCal takes another parameter along with ViSession, hence I could not just use it generally in service_helpers.mako

Reply:
Oh clever, injecting the one line close lambda like this.
I'd like to see what others think but I think something like this is fine over just custom implementing the method.

If we do go down this route, I think it would be cleaner to just have it be
```
"close_function" : "CloseExtCal(id, NIDMM_EXTCAL_ACTION_ABORT)",
```
instead to just capture the unique close function + parameters. Then it can be injected in the right spot of the cleanup lambda since the rest of the lambda matches the default cleanup lambda that gets generated so it can remain in the mako instead.

Reply:
I agree with @reckenro about not including the C++ lambda syntax. That exposes some low-level implementation details. If we put C++ code into the metadata, the key needs to reflect that. I would recommend `'custom_close' : 'CloseExtCal(id, ...)'`.